tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -116,8 +116,11 @@ def __init__(self,
         reserved_memory = max(1536, int(0.07 * node_memory) + 1)
```

Comment:
So this change ensures we drop the memory to the minimum required for the given `partition_limit` (which I guess is 1000 here)? That seems good. 

H/e we should also do the reverse test where we set memory to 1GB and validate that we can create the max allowed number of partitions (1GB / 4 = ~250 currently with the 4M divisor).

Reply:
> H/e we should also do the reverse test where we set memory to 1GB and validate that we can create the max allowed number of partitions (1GB / 4 = ~250 currently with the 4M divisor).

That's something for later though.

Reply:
I'm not sure what you're proposing here. Adding a new check in the existing test or creating a new test?

Reply:
Yeah sorry I wasn't clear enough that I wasn't suggesting it as part of this change. We should do it as part of the partition density work.